palm/plugins/dbt/parsers/project_parser.py

```diff
@@ -0,0 +1,20 @@
+class Project:
+    """Project parser model"""
+
+    def __init__(self, **data):
+        self.name = data['name']
+        self.version = data['version']
+        self.profile = data['profile']
+        self.config_version = data['config-version']
+        self.model_paths = data.get('model-paths', 'models')
+        self.macro_paths = data.get('macro-paths', 'macros')
+        self.seed_paths = data.get('seed-paths', 'seeds')
+        self.snapshot_paths = data.get('snapshot-paths', 'snapshots')
+        self.analysis_paths = data.get('analysis-paths', 'analysis')
+        self.test_paths = data.get('test-paths', 'tests')
+        self.packages_install_path = data.get('packages-install-path', 'packages')
+        self.docs_paths = data.get('docs-paths', [])
+        self.vars = data.get('vars', {})
+        self.seeds = data.get('seeds', {})
+        self.models = data.get('models', {})
+        self.snapshots = data.get('snapshots', {})
```

Comment:
Can we import `dbt` as a package? They have a dataclass for interfacing with the project config.

This is the full list of options:
```
    project_name: str
    version: Union[SemverString, float]
    project_root: str
    profile_name: Optional[str]
    model_paths: List[str]
    macro_paths: List[str]
    seed_paths: List[str]
    test_paths: List[str]
    analysis_paths: List[str]
    docs_paths: List[str]
    asset_paths: List[str]
    target_path: str
    snapshot_paths: List[str]
    clean_targets: List[str]
    log_path: str
    packages_install_path: str
    quoting: Dict[str, Any]
    models: Dict[str, Any]
    on_run_start: List[str]
    on_run_end: List[str]
    dispatch: List[Dict[str, Any]]
    seeds: Dict[str, Any]
    snapshots: Dict[str, Any]
    sources: Dict[str, Any]
    tests: Dict[str, Any]
    vars: VarProvider
    dbt_version: List[VersionSpecifier]
    packages: Dict[str, Any]
    manifest_selectors: Dict[str, Any]
    selectors: SelectorConfig
    query_comment: QueryComment
    config_version: int
    unrendered: RenderComponents
    project_env_vars: Dict[str, Any]
```

Reply:
I don't want to include dbt as a dependency, it's a very large library and we'd have a hellish time managing which version to include for the host project...

Reply:
I did consider using a dataclass... I actually implemented this with pydantic initially but that didn't make a whole lot of sense because the contents of many of the project configs are optional and the Dicts in particular have no pre-defined internal structure